### DISCUSSION TITLE
spécification pour une ligne supprimée

### DISCUSSION THREAD

**Discussion by jfcouchot** [danymat/INSAnonym-utils#66] [Q&A, Bug report & Support]
Bonjour.
Lorsqu'on supprime une ligne faut-il : 
- uniquement DEL à la place de l'id de l'utilisateur et laisser le reste de la ligne vide
- mettre DEL et \t\t\t 
- mettre DEL, modifier à la marge l'heure (rester dans la même semaine) et mettre des coordonnées aléatoires 
Merci.
femtorange

**Comment by jfcouchot** [comment]
Je pense que cela a des conséquences pour le script dateUtil
"Status : Error in processing utility script: dateUtil"

**Comment by danymat** [comment]
Bonjour, il faut uniquement mettre DEL à la place de l'id de l'utilisateur en effet. Donc le point 1 est le bon.
Essayez au maximum de tourner les métriques en local, afin d'avoir plus de détails concernant l'erreur: https://github.com/danymat/INSAnonym-utils/blob/main/docs/insanonym-metrics.md

**Reply by jfcouchot** [reply]
Rebonjour.
Je ne comprends pas trop comment fonctionne la comparaison de semaine sur dateUtil :
1) Dans runner, on ne conserve de df_anon que les lignes sans DEL
df_anon = df_anon.loc[df_anon['id'] != "DEL"]
2) Dans dateUtil, on calcule les semaines 
 df_week = df['date'].dt.isocalendar().week 
(3, 10) (11, 10) (14, 10) (17, 10) (20, 10) (23, 10) (24, 10) (26, 10) (27, 10) (30, 10) (32, 10) (35, 10) (36, 10) (37, 10) (41, 10) (42, 10) (47, 10) (49, 10) (50, 10) (56, 10) (59, 10) (66, 10) (74, 10) (76, 10) (77, 10) (80, 10) (83, 10) (84, 10) (85, ...
df_original_week = df_original_reduced['date'].dt.isocalendar().week
#(0, 10) (1, 10) (2, 10) (3, 10) (4, 10) (5, 10) (6, 10) (7, 10) (8, 10) (9, 10) (10, 10) (11, 10) (12, 10) (13, 10) (14, 10) (15, 10) (16, 10) (17, 10) (18, 10) (19, 10) (20, 10) (21, 10) (22, 10) (23, 10) (24, 10) (25, 10) (26, 10) (27, 10) (28, 10) (29, ...
3) Mais vouloir comparer ces deux séries lève une erreur :+1: 
            raise ValueError("Can only compare identically-labeled Series objects")
Les indices ne sont pas les même en effet dans les deux séries puisqu'on a supprimé des lignes dans df_anon
Une solution serait de ne pas supprimer les lignes avec  DEL pour cette métrique 
Merci pour votre aide

**Reply by beng-git** [reply]
Une petite précision que m'a expliqué Daniel : dans le jeu de données anonymisé qui est généré par la plateforme, les lignes avec DEL sont bien supprimées. Le jeu de données ainsi produit est plus petit.

**Reply by beng-git** [reply]
C'était une réponse au début du thread et non au problème plus récent :)

**Reply by danymat** [reply]
Alors je reviens sur le DEL:
si vous mettez DEL sur l'indentifiant, cela ne sert à rien de modifier le reste des colonnes, au contraire cela peut possiblement être mal interprété à la lecture initiale du fichier, si les formats ne sont pas bons

**Reply by danymat** [reply]
> Rebonjour.
> Je ne comprends pas trop comment fonctionne la comparaison de semaine sur dateUtil :
> 
> 1. Dans runner, on ne conserve de df_anon que les lignes sans DEL
>    df_anon = df_anon.loc[df_anon['id'] != "DEL"]
> 2. Dans dateUtil, on calcule les semaines
>    df_week = df['date'].dt.isocalendar().week
>    (3, 10) (11, 10) (14, 10) (17, 10) (20, 10) (23, 10) (24, 10) (26, 10) (27, 10) (30, 10) (32, 10) (35, 10) (36, 10) (37, 10) (41, 10) (42, 10) (47, 10) (49, 10) (50, 10) (56, 10) (59, 10) (66, 10) (74, 10) (76, 10) (77, 10) (80, 10) (83, 10) (84, 10) (85, ...
>    df_original_week = df_original_reduced['date'].dt.isocalendar().week
>    #(0, 10) (1, 10) (2, 10) (3, 10) (4, 10) (5, 10) (6, 10) (7, 10) (8, 10) (9, 10) (10, 10) (11, 10) (12, 10) (13, 10) (14, 10) (15, 10) (16, 10) (17, 10) (18, 10) (19, 10) (20, 10) (21, 10) (22, 10) (23, 10) (24, 10) (25, 10) (26, 10) (27, 10) (28, 10) (29, ...
> 3. Mais vouloir comparer ces deux séries lève une erreur 👍
>    raise ValueError("Can only compare identically-labeled Series objects")
>    Les indices ne sont pas les même en effet dans les deux séries puisqu'on a supprimé des lignes dans df_anon
>    Une solution serait de ne pas supprimer les lignes avec  DEL pour cette métrique
>    Merci pour votre aide

Alors je viens de voir sur le serveur:
- La dataframe du fichier original passé en paramètre est le dataframe original réduit: c'est à dire que l'on supprime les lignes où il y a DEL sur le fichier anonymisé

**Reply by danymat** [reply]
Voici le fix sur le runner: 4a7ca00. Vous pouvez pull la branche main.
Tenez moi au courant!

**Reply by beng-git** [reply]
Petite remarque : si j'ai bien compris ce que m'a dit Daniel, il n'y avait pas le 2e bug sur le code déployé sur le serveur (erreur de calcul des séries temporelles).

**Reply by jfcouchot** [reply]
Le patch proposé me semble un peu étrange
Ne s'agit-il pas plutôt de 

 if args.script in ["utility_tuile", "utility_POI","dateUtil","hourUtil"]:
        df_origin = df_origin.loc[df_anon.index]

**Reply by danymat** [reply]
Non: on a besoin du fichier original réduit sur utility_distance, dateUtil, hourUtil. Le reste a besoin du fichier original non réduit. En gros je réduis le fichier original sur toutes les métriques sauf utility_tuile, POI et meet

**Reply by jfcouchot** [reply]
C'est clair. MErci.
